tests/unit/classifier/svm/LibLinear_unittest.cc

```diff
@@ -6,1292 +6,469 @@
+		ll->set_liblinear_solver_type(liblinear_solver_type);
+		ll->train();
+		pred = ll->apply_binary(test_feats);
 
-	SG_REF(ground_truth);
-	SG_REF(train_feats);
-}
+		liblin_accuracy = eval->evaluate(pred, ground_truth);
 
-//Generate Data for L2 regularized
-void generate_data_l2(CDenseFeatures<float64_t>* &train_feats,
-					 CDenseFeatures<float64_t>* &test_feats,
-					 CBinaryLabels* &ground_truth)
-{
-	index_t num_samples = 50;
-	CMath::init_random(5);
-	SGMatrix<float64_t> data =
-		CDataGenerator::generate_gaussians(num_samples, 2, 2);
-	CDenseFeatures<float64_t> features(data);
-
-	SGVector<index_t> train_idx(num_samples), test_idx(num_samples);
-	SGVector<float64_t> labels(num_samples);
-	for (index_t i = 0, j = 0; i < data.num_cols; ++i)
-	{
-		if (i % 2 == 0)
-			train_idx[j] = i;
-		else
-			test_idx[j++] = i;
+		EXPECT_NEAR(liblin_accuracy, 1.0, 1e-6);
 
-		labels[i/2] = (i < data.num_cols/2) ? 1.0 : -1.0;
+		SG_UNREF(ll);
+		SG_UNREF(eval);
+		SG_UNREF(pred);
 	}
+	void train_with_solver_simple(
+		LIBLINEAR_SOLVER_TYPE llst, bool biasEnable, bool l1, SGVector<float64_t> t_w, bool C_value=false) //C_value only for L2R_L1LOSS_SVC_DUAL
+	{
+		LIBLINEAR_SOLVER_TYPE liblinear_solver_type = llst;
 
-	train_feats = (CDenseFeatures<float64_t>*)features.copy_subset(train_idx);
-	test_feats =  (CDenseFeatures<float64_t>*)features.copy_subset(test_idx);
-	ground_truth = new CBinaryLabels(labels);
-	SG_REF(ground_truth);
-}
+		if (l1)
+				generate_data_l1_simple();
+		else
+				generate_data_l2_simple();
 
-//Generate Data for L1 regularized,
-//it's data need to be transposed first
-void generate_data_l1_simple(CDenseFeatures<float64_t>* &train_feats,
-					 CDenseFeatures<float64_t>* &test_feats,
-					 CBinaryLabels* &ground_truth)
-{
-	index_t num_samples = 10;
-	SGMatrix<float64_t> train_data(10, 2);
-	SGMatrix<float64_t> test_data(2, 10);
-
-	train_data(0,0)=-1;	train_data(0,1)=1;
-	train_data(1,0)=-1;	train_data(1,1)=0;
-	train_data(2,0)=-1;	train_data(2,1)=-1;
-	train_data(3,0)=-2;	train_data(3,1)=0;
-	train_data(4,0)=0;	train_data(4,1)=-1;
-	train_data(5,0)=2;	train_data(5,1)=2;
-	train_data(6,0)=3;	train_data(6,1)=1;
-	train_data(7,0)=3;	train_data(7,1)=2;
-	train_data(8,0)=3;	train_data(8,1)=3;
-	train_data(9,0)=4;	train_data(9,1)=2;
-
-	test_data(0,0)=-1;	test_data(1,0)=1;
-	test_data(0,1)=-1;	test_data(1,1)=0;
-	test_data(0,2)=-1;	test_data(1,2)=-1;
-	test_data(0,3)=-2;	test_data(1,3)=0;
-	test_data(0,4)=0;	test_data(1,4)=-1;
-	test_data(0,5)=2;	test_data(1,5)=2;
-	test_data(0,6)=3;	test_data(1,6)=1;
-	test_data(0,7)=3;	test_data(1,7)=2;
-	test_data(0,8)=3;	test_data(1,8)=3;
-	test_data(0,9)=4;	test_data(1,9)=2;
-
-	train_feats = new CDenseFeatures<float64_t>(train_data);
-	test_feats = new CDenseFeatures<float64_t>(test_data);
-
-	SGVector<float64_t> labels(num_samples);
-	for (index_t i = 0; i < num_samples; ++i)
-	{
-		labels[i] = (i < num_samples/2) ? 1.0 : -1.0;
-	}
+		CLibLinear* ll = new CLibLinear();
```

Comment:
same as above... this code shares quite a lot of lines with `train_with_solver`... we could combine them :)

Reply:
I did think about this, but when I did it, it had a lot of if statements and it wasn't very readable (for a test case). Are we sure we wanna do this?